### DISCUSSION TITLE
Will not boot and menues glitching

### DISCUSSION THREAD

**Discussion by dennishusell** [OnionUI/Onion#1057] [Support] [answered]
### Checklist

- [X] I am reporting a bug in the Onion OS - Any issues with the closed source frontend ([MainUI](https://github.com/OnionUI/MainUI-issues)), or [RetroArch](https://github.com/libretro/RetroArch) or any of the cores should be directed to their respective repositories
- [X] I have reviewed the [Onion Wiki](https://github.com/OnionUI/Onion/wiki) and did not find the answer
- [X] I have searched the existing (open and closed) issues to ensure I do not log a duplicate

### Onions OS Version

4.2.0

### Miyoo Firmware Version

202305052130

### Provide a clear and concise description of the issue

I was adding some new ROMs to my SD card and pulled it out while the device was running (on the main menu).
I did this several times back and forth, but the final time my device went into sleep during the time it did not have an SD card installed.
After finishing the transfer i reinstalled the SD card and tried to boot up the device, however it only showed the Miyoo bootup screen and then went black (power LED was green), no matter what button or combo i pushed i could not get it to power up completely. I even tried pulling the battery but it was the same issue.
When the device was stuck at the black screen, removing and reinstalling the SD card would cause it to finish booting up properly, however it would be glitchy, randomly closing down games very quickly after starting, some menues such as SELECT button not being present etc.

### Steps to Reproduce

1. Remove SD card while on the main menu.
2. Transfer some new ROMs to SD card.
3. Let the device fall asleep with SD card removed.
4. Reinstall SD card and try to boot up.

### Provide any additional context or information that may help us investigate your issue

What i did that fixed it (Some parts may be redundant, but this is exactly what i did):
1. Boot up device with original SD card/OS.
2. Played a random game for a few seconds.
3. Shut down the device.
4. Transfer some files to the OnionOS SD card.*
5. Boot up device with the OnionOS SD card.

*In my case, i added a new folder on the root called OnionOS Installer, as i was planning on bringing the device to work next morning and trying a complete re-install.

**Comment by Aemiii91** [comment, marked as answer]
Removing the card while the device is on is bound to cause file corruption. You need to reformat the card and stop removing it unsafely.
